packages/runtime/container-runtime/src/dataStoreContexts.ts

```diff
@@ -6,13 +6,50 @@
 import { IDisposable, ITelemetryBaseLogger, ITelemetryLogger } from "@fluidframework/common-definitions";
 import { assert, Deferred, Lazy } from "@fluidframework/common-utils";
 import { ChildLogger } from "@fluidframework/telemetry-utils";
+import { DataProcessingError } from "@fluidframework/container-utils";
 import { FluidDataStoreContext, LocalFluidDataStoreContext } from "./dataStoreContext";
 
+export class TombStonableDataStoreContexts extends Map<string, FluidDataStoreContext> {
+    private readonly contexts = new Map<string, FluidDataStoreContext>();
+    private readonly tombstonedContexts = new Set<string>();
+    public tombstone(id: string) {
+        this.tombstonedContexts.add(id);
+    }
+
+    private isTombstoned(id: string) {
+        return this.tombstonedContexts.has(id);
+    }
+
+    private validateNotTombstoned(id: string) {
+        if (this.isTombstoned(id)) {
+            throw DataProcessingError.wrapIfUnrecognized(
+                new Error(`Datastore: ${id} has been tombstoned!`), "tombstonedGarbageCollectionDatastoreAccessed");
+        }
+    }
+
+    delete(key: string): boolean {
+        return this.contexts.delete(key);
+    }
+
+    get(key: string): FluidDataStoreContext | undefined {
```

Comment:
This does seem like a good, centralized place to add this validation. However, throwing an error every time get / has is called is not a good idea. get can be called for any number of reasons and we don't want to simply crash when that happens.
Also, this will not prevent the communication from data store -> container runtime.